hls4ml/templates/vivado/nnet_utils/nnet_recr_activations.h

```diff
@@ -0,0 +1,63 @@
+#ifndef NNET_RECR_ACTIVATION_H_
+#define NNET_RECR_ACTIVATION_H_
+
+#include "nnet_common.h"
+#include "nnet_helpers.h"
+#include "nnet_activation.h"
+#include "hls_stream.h"
+#include <math.h>
+
+namespace nnet {
+
+namespace activation{
+
+template<class data_T, class res_T, typename CONFIG_T>
+class Activation{
+    public:
+    // *************************************************
+    //       Blank Activation
+    // *************************************************
+    static void activation(data_T data[CONFIG_T::n_in], res_T res[CONFIG_T::n_in]) {} // Nothing to do here
+};
+
+template<class data_T, class res_T, typename CONFIG_T>
+class relu : public Activation<data_T, res_T, CONFIG_T>{
+    public:
+    // *************************************************
+    //       Relu Activation
+    // *************************************************
+    static void activation(data_T data[CONFIG_T::n_in], res_T res[CONFIG_T::n_in])
+    {
+        nnet::relu<data_T, res_T, CONFIG_T>(data, res);   
```

Comment:
Out of curiosity, does Vivado inline this call automatically, or can we save a cycle here with a simple pragma?